src/shims/fs.rs

```diff
@@ -139,6 +139,23 @@ pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriEvalContextExt<'mir, 'tcx
             } else {
                 this.handle_not_found()
             }
+        } else if cmd == this.eval_libc_i32("F_DUPFD")? || cmd == this.eval_libc_i32("F_DUPFD_CLOEXEC")? {
+            let arg = match arg_op {
+                Some(arg_op) => this.read_scalar(arg_op)?.to_i32()?,
+                None => throw_unsup_format!("fcntl with command F_DUPFD or F_DUPFD_CLOEXEC requires a third argument"),
```

Comment:
I tried calling this without `arg` and to the best of my knowledge it is equivalent  to having `arg = 0` (probably because the value is just `NULL`?). However this does not seem to be documented anywhere.

Reply:
`man fcntl` on my system says

> fcntl() can take an optional third argument.  Whether or not this argument  is  required is determined by cmd.  The required argument type is indicated in parentheses after each cmd name (in most  cases,  the  required  type  is int, and we identify the argument using the name arg), or void is specified if the argument is not required.

Then, further down, F_DUPFD and F_DUPFD_CLOEXEC both say they take an int for the third argument.

Reply:
Yes I agree on that, calling `F_DUPFD` without its argument should be an error, however the actual behavior is different. I'm thorn between the two options.

Reply:
If the docs say it is an error, then it should error in Miri. The implementation in Linux might just do less error checking than it could, but we should do the maximal amount of error checking we can.